cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -99,6 +100,9 @@ func NewCommand() *cobra.Command {
 			errors.CheckError(err)
 
 			restConfig.UserAgent = fmt.Sprintf("argocd-applicationset-controller/%s (%s)", vers.Version, vers.Platform)
+			if acceptProtobufContentType {
+				restConfig.AcceptContentTypes = runtime.ContentTypeProtobuf + "," + runtime.ContentTypeJSON
+			}
```

Comment:
Should this be a part of getting ClientConfig()? Overriding it here seems a bit hacky.